### PR TITLE
Static analysis

### LINKED ISSUE #1: Manuális átvizsgálás, statikus analízis
Manuális kód átvizsgálás elvégzése az alkalmazás egy részére (GitHub, Gerrit...) + Statikus analízis eszköz futtatása és jelzett hibák átnézése (SonarCloud, SpotBugs, VS Code Analyzer, Codacy, Coverity Scan...).
Mivel az eszközök rengeteg hibát és figyelmeztetést találhatnak, ezért elég azok egy részét megvizsgálni és ha a csapat minden tagja egyetért vele, akkor javítani. Törekedjetek arra, hogy különböző típusú, és lehetőleg nem triviális hibajelzéseket vizsgáljatok meg.

### PR DESCRIPTION
**Pull request by szannaa**
Unit tests still pass

### CODE REVIEW

**Comment by szannaa** [comment]
Should I fix the ones I didnt cross out?

File: documentation/I7F4U6/i7f4u6.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Mivel azon a projlab projekten dolgozunk, aminek az elkészítésében én is részt vettem, a feladataimat a projekt azon részén végzem el, amiket én ítam.
```
Comment (adambakurecz) [inline documentation/I7F4U6/i7f4u6.md]: nit: írtam

File: documentation/I7F4U6/i7f4u6.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Manuális kód átvizsgálás
+
+## Character
+
+A Character osztály a játékban szereplő karakterek absztrakt reprezentációja. Megvalósítja a Nameable és a Comparable<Character> interfészeket, és különféle módszereket biztosít a játék környezetével való interakcióra, például mezők közötti mozgásra, csövekkel és pumpákkal való interakcióra, valamint különböző állapotok kezelésére, mint például a beragadás.
+
+**Észrevételek:**
+1. <del>Helykitöltő metódusokat meg lehetne jelölni absztraktként és az alosztályokban megvalósítani őket<del>
+2. A compareTo metódus nem túl oop, de túl sok dolog épít rá a projekt többi részében
```
Comment (adambakurecz) [inline documentation/I7F4U6/i7f4u6.md]: `compareTo` should be fine like this.

File: documentation/I7F4U6/i7f4u6.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+3. Az equals metódus sem túl oop, de túl sok dolog épít rá a projekt többi részében
```
Comment (adambakurecz) [inline documentation/I7F4U6/i7f4u6.md]: Same goes for `equals`.

File: documentation/I7F4U6/i7f4u6.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+4. <del>Typo: In word 'recieve'<del>
+
+## MechanicCharacter
+
+A MechanicCharacter osztály egy speciális karaktert reprezentál a játékban, aki képes csöveket és pumpákat javítani vagy állítani. Az osztály kiterjeszti a Character osztályt, és új funkcionalitást ad hozzá a játékhoz.
+
+**Észrevételek:**
+1. <del>Lehetne javítani a toString metóduson, hogy olvashatóbb legyen<del>
+2. A null ellenőrzések néha !=, néha ==-vel történnek, ami fejlesztés közben is problémát okozott az átláthatóságban
```
Comment (adambakurecz) [inline documentation/I7F4U6/i7f4u6.md]: Since functionally it's the same, we can let it slide.

**Review by adambakurecz — APPROVED** [review]
The sonarlint files should be included in `.gitignore` but other than that LGTM.

**Comment by szannaa** [comment]
SonarCloud Code Analysis should be fine after merging to main. If not reopen this issue.

**Review by saviourdavid — APPROVED** [review]
"Using SonarCloud is like having a hyperactive grammar teacher for your code—constantly pointing out your mistakes, but in the end, making you look like a genius!"

-GPT 4.0

**Pull request merged**